executor/executor.cc

```diff
@@ -1146,12 +1146,21 @@ void copyin_int(char* addr, uint64 val, uint64 bf, uint64 bf_off, uint64 bf_len)
 		return;
 	}
 	T x = swap(*(T*)addr, sizeof(T), bf);
-	x = (x & ~BITMASK(bf_off, bf_len)) | ((val << bf_off) & BITMASK(bf_off, bf_len));
+	debug("copyin_int<%d>: old x=0x%llx\n", (int)sizeof(T), (uint64)x);
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+	const uint64 shift = sizeof(T) * 8 - bf_off - bf_len;
```

Comment:
If we change this, I assume we also need to change STORE_BY_BITMASK for the C reproducers (and/or how pkg/csource uses STORE_BY_BITMASK b/c it knows endianness statically and could do swap itself).

However, this should have need caught by pkg/runtest tests as they also generate and run and check C reproducers...